crates/control/src/motion/walking_engine.rs

```diff
@@ -186,9 +197,11 @@ impl WalkingEngine {
             WalkState::Kicking(..) => self.kick_cycle(last_cycle_duration),
         }
 
-        let has_support_changed = match context.support_foot.support_side {
-            Some(support_side) => self.swing_side.opposite() != support_side,
-            None => true,
+        let left_foot_pressure = context.sensor_data.force_sensitive_resistors.left.sum();
+        let right_foot_pressure = context.sensor_data.force_sensitive_resistors.right.sum();
+        let has_support_changed = match self.swing_side {
+            Side::Left => left_foot_pressure > right_foot_pressure,
+            Side::Right => right_foot_pressure > left_foot_pressure,
```

Comment:
Citing myself from the issue #38 : `But from lots of testing a hysterese should only be necessary if the FSRs of both feet measure close to 0 pressure`.  
This version here will probably work. But for us I noticed that without some hysterese (like a minimum foot pressure) around the 0 mark, meaning both feet measure close to nothing, the robot will walk in specific situations "not that good". If I remember correctly, those situations are when the robot is walking on his heels, e.g. after a push from the front which tilts the robot backwards. Also when turning in place (meaning walk steps with a lot of rotation, but no translation) the FSRs like to measure not that much, which caused too early foot support switches. 

Another thing I noticed is that most teams use weights for the FSRs (inner pressures are weighted different than outter ones). We use the same values rUNSWift provide, but never tested if those make a difference. Just to have it mentioned in case it could improve the decision for a support foot switch.

Reply:
Thanks for your remarks! We have a minimal_step_duration for each step, which works like a hysterese for preventing too early support foot switches. If the FSRs values are close to zero, because we lose ground contact, our has_ground_contact check prevents the robot from unwanted foot switches.

We will look into the benefits for using weights for the FSRs in the next iteration.